lib/3scale/backend/user.rb

```diff
@@ -82,10 +82,29 @@ def self.delete!(service_id, username)
         storage.del(self.key(service_id, username))
       end
 
+      def self.delete_all(service_id)
+        service = Service.load_by_id(service_id)
+        raise UserRequiresValidService if service.nil?
+        current_scan_cursor_position = "0"
+        finished = false
+        users_set_key = service_users_set_key(service_id)
+        while !finished
+          (current_scan_cursor_position, current_usernames) =
+            storage.sscan(users_set_key, current_scan_cursor_position,
+                          { count: PIPELINED_SLICE_SIZE })
+          delete_users_in_batch(service_id,  current_usernames)
+          finished = current_scan_cursor_position == "0"
```

Comment:
can you assure that finished will be eventually ```true``` , no matter what? on redis connection error, exception will rise, right?